documents/architecture-decisions.md

```diff
@@ -751,28 +759,29 @@ In the state `Approved` the Publisher has 2 options:
 It should be possible to add comments to the CSAF document. The comment could be
 for the whole document or for a specific area in the document.
 Since the CSAF standard has no concept for unique identifiers inside the
-document we need to persist this relation somehow without unnecessarily adding
-identifiers to each object.
+document such identifiers are added to each object node to persist this relation.
 
 #### Decision
 
-The ids of the Comments are referenced from the CSAF Document objects.
-When the comment belongs to a dedicated field and not the whole object,
-the fieldName in the objects is used to specify the concrete value.
-
-We archive this by adding a $comment value to the document where the user adds a
-comment.
+These missing unique identifiers are added to each object node in the CSAF JSON.
+A comment then refers to such a node ID.
+When the comment belongs to a dedicated field and not the whole object, the
+fieldName in the objects is used to specify the concrete value.
 
 #### Consequences
 
-- The algorithm to add comments is very simple
-- The comments are referenced proper even some parts of the document are deleted
-- The size of the CSAF document is only slightly increased be the comments
-- The comments have to be removed before the CSAF document is validated
-- The comments have to be removed / ignored when exporting the document
-- The rest client has to manage the id of the comments
-- The creation of comments and saving the CSAF documents is done in different
-  transaction. We need a job cleanup for accidentally created comments
+- The algorithm to add comments is very simple, they are just persisted as new
+  documents in the database
+- The CSAF document does not change when a comment is created
+- The node IDs have to be removed before the CSAF document is validated
```

Comment:
The CSAF document with the uuid has to be saved before comments can be added.

The names should be consistent uuid/nodeId.

Reply:
updated